src/main/java/appeng/helpers/NonNullArrayIterator.java

```diff
@@ -49,8 +49,9 @@ public boolean hasNext()
 	@Override
 	public E next()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a "NoSuchElementException" for iteration beyond the end of the collection. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2272)